launch/launch/substitutions/this_launch_file_dir.py

```diff
@@ -14,20 +14,31 @@
 
 """Module for the ThisLaunchFileDir substitution."""
 
+from typing import Iterable
 from typing import Text
 
 from .substitution_failure import SubstitutionFailure
+from ..frontend.expose import expose_substitution
 from ..launch_context import LaunchContext
+from ..some_substitutions_type import SomeSubstitutionsType
 from ..substitution import Substitution
 
 
+@expose_substitution('dirname')
 class ThisLaunchFileDir(Substitution):
     """Substitution that returns the absolute path to the current launch file."""
 
     def __init__(self) -> None:
         """Constructor."""
         super().__init__()
 
+    @classmethod
+    def parse(cls, data: Iterable[SomeSubstitutionsType]):
+        """Parse `EnviromentVariable` substitution."""
+        if len(data) > 1:
```

Comment:
@ivanpauno shouldn't it be
```suggestion
        if len(data) != 0:
```
?

Reply:
Corrected in https://github.com/ros2/launch/pull/273/commits/dc75c2375e7911624a16b08577e6aa3ec5d8050a.